execution/native/precompiles.go

```diff
@@ -34,29 +39,33 @@ func leftPadAddress(bs ...byte) crypto.Address {
 	return crypto.AddressFromWord256(binary.LeftPadWord256(bs))
 }
 
-/* Removed due to C dependency
-func ecrecoverFunc(state State, caller crypto.Address, input []byte, gas *int64) (output []byte, err error) {
+func ecrecoverFunc(ctx Context) (output []byte, err error) {
 	// Deduct gas
 	gasRequired := GasEcRecover
-	if *gas < gasRequired {
-		return nil, ErrInsufficientGas
+	if *ctx.Gas < gasRequired {
+		return nil, errors.Codes.InsufficientGas
 	} else {
-		*gas -= gasRequired
+		*ctx.Gas -= gasRequired
 	}
-	// Recover
-	hash := input[:32]
-	v := byte(input[32] - 27) // ignore input[33:64], v is small.
-	sig := append(input[64:], v)
 
-	recovered, err := secp256k1.RecoverPubkey(hash, sig)
+	// SECP256K1 Recovery
+	hash := ctx.Input[:32] // from 0 to 31 bytes
+	sig := ctx.Input[63:]  // skip values in range[32:62]
+
+	publicKey, isCompressed, err := ecdsa.RecoverCompact(sig, hash)
```

Comment:
Please our existing secp256k1-providing dependency `"github.com/btcsuite/btcd/btcec"`: `btcec.RecoverCompact` should do the job.